File: frontend/src/api/workspaceIntelligence.ts
import { apiDelete, apiGet, apiPost, apiPut } from './client';

export type WorkspaceSourceType = 'website' | 'facebook_page' | 'instagram_account' | 'linkedin_page' | 'whatsapp_business' | 'manual';

export interface WorkspaceProfile {
  id: number;
  organization_id: number;
  business_description?: string | null;
  mission?: string | null;
  tagline?: string | null;
  industry?: string | null;
  services: string[];
  products: string[];
  target_audience?: string | null;
  locations: string[];
  brand_voice?: string | null;
  tone?: string | null;
  visual_style?: string | null;
  brand_colors: string[];
  font_preferences: string[];
  preferred_content_formats: string[];
  content_cadence: Record<string, unknown>;
  keywords: string[];
  preferred_languages: string[];
  contact_email?: string | null;
  contact_phone?: string | null;
  whatsapp_display_phone?: string | null;
  whatsapp_business_account_id?: string | null;
  website_url?: string | null;
  linkedin_url?: string | null;
  facebook_url?: string | null;
  instagram_url?: string | null;
  whatsapp_url?: string | null;
  logo_media_id?: number | null;
  telegram_approval_chat_id?: string | null;
  telegram_approval_user_id?: string | null;
  telegram_approval_enabled: boolean;
  approval_required: boolean;
  approved_claims: string[];
  prohibited_claims: string[];
  last_refreshed_at?: string | null;
  created_at: string;
  updated_at?: string | null;
}

export interface WorkspaceSource {
  id: number;
  organization_id: number;
  source_type: WorkspaceSourceType;
  provider?: string | null;
  url?: string | null;
  external_id?: string | null;
  title?: string | null;
  content_text?: string | null;
  excerpt?: string | null;
  metadata: Record<string, unknown>;
  trust_level: string;
  review_status: string;
  is_active: boolean;
  last_fetched_at?: string | null;
  created_at: string;
  updated_at?: string | null;
}

export interface ContentOpportunity {
  id: number;
  organization_id: number;
  source_type: string;
  source_url?: string | null;
  publisher?: string | null;
  external_id?: string | null;
  title: string;
  summary?: string | null;
  source_published_at?: string | null;
  discovered_at: string;
  freshness_score: number;
  relevance_score: number;
  trust_score: number;
  status: string;
  metadata: Record<string, unknown>;
}

export interface WorkspaceIntelligence {
  profile: WorkspaceProfile | null;
  sources: WorkspaceSource[];
  source_count: number;
  approved_source_count: number;
}

export type WorkspaceProfileInput = Omit<WorkspaceProfile, 'id' | 'organization_id' | 'last_refreshed_at' | 'created_at' | 'updated_at'>;

export interface WorkspaceSourceInput {
  source_type: WorkspaceSourceType;
  provider?: string;
  url?: string;
  external_id?: string;
  title?: string;
  content_text?: string;
  excerpt?: string;
  metadata?: Record<string, unknown>;
  trust_level?: string;
  review_status?: string;
}

export function listContentOpportunities(orgId: number): Promise<ContentOpportunity[]> {
  return apiGet<ContentOpportunity[]>(`organizations/${orgId}/intelligence/opportunities`);
}

export function discoverContentOpportunities(orgId: number): Promise<ContentOpportunity[]> {
  return apiPost<ContentOpportunity[]>(`organizations/${orgId}/intelligence/opportunities/discover`);
}

export function getWorkspaceIntelligence(orgId: number): Promise<WorkspaceIntelligence> {
  return apiGet<WorkspaceIntelligence>(`organizations/${orgId}/intelligence`);
}

export function saveWorkspaceProfile(orgId: number, payload: WorkspaceProfileInput): Promise<WorkspaceProfile> {
  return apiPut<WorkspaceProfile>(`organizations/${orgId}/intelligence/profile`, payload);
}

export function addWorkspaceSource(orgId: number, payload: WorkspaceSourceInput): Promise<WorkspaceSource> {
  return apiPost<WorkspaceSource>(`organizations/${orgId}/intelligence/sources`, payload);
}

export function refreshWorkspaceSource(orgId: number, sourceId: number): Promise<WorkspaceSource> {
  return apiPost<WorkspaceSource>(`organizations/${orgId}/intelligence/sources/${sourceId}/refresh`);
}

export function reviewWorkspaceSource(orgId: number, sourceId: number, reviewStatus: 'pending' | 'approved' | 'rejected', reviewNote?: string): Promise<WorkspaceSource> {
  return apiPost<WorkspaceSource>(`organizations/${orgId}/intelligence/sources/${sourceId}/review`, { review_status: reviewStatus, review_note: reviewNote });
}

export function refreshWorkspaceSources(orgId: number): Promise<{ refreshed_source_ids: number[]; errors: string[]; refreshed_count: number }> {
  return apiPost<{ refreshed_source_ids: number[]; errors: string[]; refreshed_count: number }>(`organizations/${orgId}/intelligence/refresh`);
}

export function removeWorkspaceSource(orgId: number, sourceId: number): Promise<void> {
  return apiDelete(`organizations/${orgId}/intelligence/sources/${sourceId}`);
}

File: app/schemas/workspace_intelligence.py
"""Schemas for workspace business intelligence and source provenance."""
from datetime import datetime
from typing import Any, Optional

from pydantic import AnyHttpUrl, BaseModel, ConfigDict, Field


class WorkspaceProfileUpsert(BaseModel):
    business_description: Optional[str] = Field(default=None, max_length=10000)
    mission: Optional[str] = Field(default=None, max_length=5000)
    tagline: Optional[str] = Field(default=None, max_length=500)
    industry: Optional[str] = Field(default=None, max_length=255)
    services: list[str] = Field(default_factory=list, max_length=100)
    products: list[str] = Field(default_factory=list, max_length=100)
    target_audience: Optional[str] = Field(default=None, max_length=5000)
    locations: list[str] = Field(default_factory=list, max_length=100)
    brand_voice: Optional[str] = Field(default=None, max_length=5000)
    tone: Optional[str] = Field(default=None, max_length=100)
    visual_style: Optional[str] = Field(default=None, max_length=5000)
    brand_colors: list[str] = Field(default_factory=list, max_length=20)
    font_preferences: list[str] = Field(default_factory=list, max_length=20)
    preferred_content_formats: list[str] = Field(default_factory=list, max_length=20)
    content_cadence: dict[str, Any] = Field(default_factory=dict)
    keywords: list[str] = Field(default_factory=list, max_length=100)
    watch_terms: list[str] = Field(default_factory=list, max_length=100)
    competitor_urls: list[AnyHttpUrl] = Field(default_factory=list, max_length=50)
    preferred_languages: list[str] = Field(default_factory=list, max_length=20)
    contact_email: Optional[str] = Field(default=None, max_length=320)
    contact_phone: Optional[str] = Field(default=None, max_length=80)
    whatsapp_display_phone: Optional[str] = Field(default=None, max_length=80)
    whatsapp_business_account_id: Optional[str] = Field(default=None, max_length=255)
    website_url: Optional[AnyHttpUrl] = None
    linkedin_url: Optional[AnyHttpUrl] = None
    facebook_url: Optional[AnyHttpUrl] = None
    instagram_url: Optional[AnyHttpUrl] = None
    whatsapp_url: Optional[AnyHttpUrl] = None
    logo_media_id: Optional[int] = Field(default=None, ge=1)
    telegram_approval_chat_id: Optional[str] = Field(default=None, max_length=255)
    telegram_approval_user_id: Optional[str] = Field(default=None, max_length=255)
    telegram_approval_enabled: bool = False
    approval_required: bool = True
    approved_claims: list[str] = Field(default_factory=list, max_length=100)
    prohibited_claims: list[str] = Field(default_factory=list, max_length=100)


class WorkspaceProfileResponse(WorkspaceProfileUpsert):
    id: int
    organization_id: int
    last_refreshed_at: Optional[datetime] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)


class WorkspaceSourceCreate(BaseModel):
    source_type: str = Field(pattern="^(website|facebook_page|instagram_account|linkedin_page|whatsapp_business|rss|news|research|manual)$")
    provider: Optional[str] = Field(default=None, max_length=50)
    url: Optional[AnyHttpUrl] = None
    external_id: Optional[str] = Field(default=None, max_length=255)
    title: Optional[str] = Field(default=None, max_length=500)
    content_text: Optional[str] = Field(default=None, max_length=200000)
    excerpt: Optional[str] = Field(default=None, max_length=10000)
    metadata: dict[str, Any] = Field(default_factory=dict)
    trust_level: str = Field(default="user_supplied", pattern="^(user_supplied|official_api|public_metadata|manual_reviewed)$")
    review_status: str = Field(default="pending", pattern="^(pending|approved|rejected)$")


class WorkspaceSourceReview(BaseModel):
    review_status: str = Field(pattern="^(pending|approved|rejected)$")
    review_note: Optional[str] = Field(default=None, max_length=2000)


class WorkspaceSourceResponse(WorkspaceSourceCreate):
    id: int
    organization_id: int
    is_active: bool
    last_fetched_at: Optional[datetime] = None
    created_at: datetime
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)


class ContentOpportunityResponse(BaseModel):
    id: int
    organization_id: int
    source_type: str
    source_url: Optional[str] = None
    publisher: Optional[str] = None
    external_id: Optional[str] = None
    title: str
    summary: Optional[str] = None
    source_published_at: Optional[datetime] = None
    discovered_at: datetime
    freshness_score: float
    relevance_score: float
    trust_score: float
    status: str
    metadata: dict[str, Any] = Field(default_factory=dict)


class WorkspaceIntelligenceResponse(BaseModel):
    profile: Optional[WorkspaceProfileResponse] = None
    sources: list[WorkspaceSourceResponse] = Field(default_factory=list)
    source_count: int
    approved_source_count: int
